cmd/root.go

```diff
@@ -27,63 +29,92 @@ type Scanner interface {
 var level logconfig.Level
 
 // print result of scan as JSON to stdout
-var printJSON bool
+var outputScan bool
 
 // list of namespaces to be skipped from scan
 var skippedNs []string
 
 // skip SSL cert validation when connecting to PolicyServers endpoints
 var insecureSSL bool
 
-var (
-	// rootCmd represents the base command when called without any subcommands
-	rootCmd = &cobra.Command{
-		Use:   "audit-scanner",
-		Short: "Reports evaluation of existing Kubernetes resources with your already deployed Kubewarden policies",
-		Long: `Scans resources in your kubernetes cluster with your already deployed Kubewarden policies.
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "audit-scanner",
+	Short: "Reports evaluation of existing Kubernetes resources with your already deployed Kubewarden policies",
+	Long: `Scans resources in your kubernetes cluster with your already deployed Kubewarden policies.
 Each namespace will have a PolicyReport with the outcome of the scan for resources within this namespace.
 There will be a ClusterPolicyReport with results for cluster-wide resources.`,
 
-		RunE: func(cmd *cobra.Command, args []string) error {
-			level.SetZeroLogLevel()
-			namespace, err := cmd.Flags().GetString("namespace")
-			if err != nil {
-				return err
-			}
-			kubewardenNamespace, err := cmd.Flags().GetString("kubewarden-namespace")
-			if err != nil {
-				return err
-			}
-			clusterWide, err := cmd.Flags().GetBool("cluster")
+	RunE: func(cmd *cobra.Command, args []string) error {
+		level.SetZeroLogLevel()
+		namespace, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			return err
+		}
+		kubewardenNamespace, err := cmd.Flags().GetString("kubewarden-namespace")
+		if err != nil {
+			return err
+		}
+		clusterWide, err := cmd.Flags().GetBool("cluster")
+		if err != nil {
+			return err
+		}
+		policyServerURL, err := cmd.Flags().GetString("policy-server-url")
+		if err != nil {
+			return err
+		}
+		caCertFile, err := cmd.Flags().GetString("extra-ca")
+		if err != nil {
+			return err
+		}
+		storeType, err := cmd.Flags().GetString("store")
+		if err != nil {
+			return err
+		}
+
+		policyReportStore, err := getPolicyReportStore(storeType)
+		if err != nil {
+			return err
+		}
+		policiesFetcher, err := policies.NewFetcher(kubewardenNamespace, skippedNs)
+		if err != nil {
+			return err
+		}
+		resourcesFetcher, err := resources.NewFetcher(kubewardenNamespace, policyServerURL)
+		if err != nil {
+			return err
+		}
+		scanner, err := scanner.NewScanner(policyReportStore, policiesFetcher, resourcesFetcher, insecureSSL, caCertFile)
+		if err != nil {
+			return err
+		}
+
+		if err := startScanner(namespace, clusterWide, scanner); err != nil {
+			return err
+		}
+
+		if outputScan {
```

Comment:
I think it should stay. The in memory store should be there just for testing. We implement the interface in the memory store or for example a future SQL backend.

Reply:
my apologies @viccuad, but I didn't understood what you meant with
> I think we should have this logic to do ToJSON and print in cmd/root.go, and it should be on scanner.go.

Reply:
I meant that I would like the block `if outputScan { scanner.ReportStore.ToJSON() }` to stay in `scanner/scanner.go` as it was before. This puts all the logic in the internal libs, and leaves `cmd/*` thinner, just dealing with flags and making the big calls. It will simplify future testing and refactoring too.

Reply:
I get your point now, thank you for your explanation @viccuad ! one thing that it's important to mention is that this way we will output all store data (either Kubernetes or Memory) multiple times per namespace which is also not ideal